nautobot/utilities/renderers.py

```diff
@@ -87,87 +72,111 @@ def validate_action_buttons(self, view, request):
+                    pk_list = [obj.pk for obj in view.filterset_class(request.GET, view.queryset.only("pk")).qs]
+                else:
+                    pk_list = request.POST.getlist("pk")
+                if "_apply" in request.POST:
+                    form = view.bulk_edit_form_class(model, request.POST)
+                    restrict_form_fields(form, request.user)
+                else:
+                    # Include the PK list as initial data for the form
+                    initial_data = {"pk": pk_list}
+                    # Check for other contextual data needed for the form. We avoid passing all of request.GET because the
+                    # filter values will conflict with the bulk edit form fields.
+                    # TODO: Find a better way to accomplish this
+                    if "device" in request.GET:
+                        initial_data["device"] = request.GET.get("device")
+                    elif "device_type" in request.GET:
+                        initial_data["device_type"] = request.GET.get("device_type")
+
+                    form = view.bulk_edit_form_class(model, initial=initial_data)
+                    restrict_form_fields(form, request.user)
+
+        context = {
             "obj": obj,
-            "object": obj,
-            "obj_type": model._meta.verbose_name,
-            "obj_type_plural": model._meta.verbose_name_plural,
-            # "form": data.serializer,
-            "form": form,
-            "return_url": self.get_return_url(request, view, obj),
+            "object": instance,
+            "obj_type": view.queryset.model._meta.verbose_name,
+            "obj_type_plural": view.queryset.model._meta.verbose_name_plural,
             "editing": obj.present_in_database,
-            "template": self.template,
-            # "serializer": data.serializer,
-            # "style": style,
+            "form": form,
+            "fields": view.import_form().fields,
+            "table": table if table else data.get("table", None),
+            "return_url": view.get_return_url(request, instance),
+            "verbose_name": view.queryset.model._meta.verbose_name,
+            "verbose_name_plural": view.queryset.model._meta.verbose_name_plural,
+            "changelog_url": view.get_changelog_url(instance),
             "content_type": content_type,
-            "table": table,
             "permissions": permissions,
             "action_buttons": valid_actions,
-            "table_config_form": TableConfigForm(table=table),
-            "filter_form": view.filterset_form(request.data, label_suffix="") if view.filterset_form else None,
-        })
-
+            "table_config_form": TableConfigForm(table=table) if table else None,
+            "filter_form": self.get_filter_form(view, request),
+            "active_tab": "csv-data",
+        }
+        if view.action == "retrieve":
+            context.update(view.get_extra_context(request, ACTION_TO_VIEW_TYPE[view.action], instance))
```

Comment:
Is this another case where we would want to just use `view.action` directly instead of remapping it?